clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -2142,9 +2142,9 @@ Value *ScalarExprEmitter::VisitInitListExpr(InitListExpr *E) {
   bool Ignore = TestAndClearIgnoreResultAssign();
   (void)Ignore;
   unsigned NumInitElements = E->getNumInits();
-  assert(Ignore == false ||
-         (NumInitElements == 0 && E->getType()->isVoidType()) &&
-             "init list ignored");
+  assert((Ignore == false ||
+          (NumInitElements == 0 && E->getType()->isVoidType())) &&
+         "init list ignored");
```

Comment:
The following alternative is perhaps more clear, and allows to remove the `(void)Ignore` line from just above.

```suggestion
  if (Ignore)
    assert(NumInitElements == 0 && E->getType()->isVoidType()
           && "only ignore irrelevant init list");
```

Reply:
OK, good to me. The extra parentheses solve the GCC warning.